packages/core/src/internal/proc.js

```diff
@@ -338,15 +338,20 @@ export default function proc(
       })
 
       if (!task.cont) {
-        if (result.sagaStack) {
+        if (result && result.sagaStack) {
           result.sagaStack = sagaStackToString(result.sagaStack)
         }
 
         if (result instanceof Error && onError) {
           onError(result)
+        } else if (onError) {
+          const errorString = result && result.toString() ?
```

Comment:
if user throws null or undefined, why we can't just pass it to `onError` as is?

Reply:
Sending an error is important to not introduce a breaking change - we have no idea how many people in the world are expecting an Error object to be passed into `onError`, and I didn't want to break their code.

Reply:
This also has the added benefit of sending up a stack trace so someone could attempt to debug what happened during the error based on that

Reply:
You are targeting at the moment v1-beta line (and I really wouldnt like to do any development on 0.x line), breaking changes can happen during a beta period and this also shouldnt break too many ppl, so it's fine to just pass what was thrown "as is" to `onError`